File: backend/Bookstore.API/Data/Book.cs
/*
    Mary Catherine Shepherd
    IS 413
    Mission 11

    Book.cs

    This class represents a Book entity in the database.
    Each property corresponds to a column in the Books table
    of the SQLite database used in the bookstore application.
*/

using System.ComponentModel.DataAnnotations;

namespace Bookstore.API.Data
{
    public class Book
    {
        // Primary key for the Books table
        [Key]
        public int BookID { get; set; }

        // Title of the book
        [Required]
        public string Title { get; set; }

        // Author of the book
        [Required]
        public string Author { get; set; }

        // Publishing company
        [Required]
        public string Publisher { get; set; }

        // ISBN number used to uniquely identify the book
        [Required]
        public string ISBN { get; set; }

        // Category or classification of the book (ex: Fiction, Non-Fiction)
        [Required]
        public string Classification { get; set; }

        // Number of pages in the book
        [Required]
        public int PageCount { get; set; }

        // Price of the book
        [Required]
        public decimal Price { get; set; }
    }
}

File: frontend/src/types/Book.ts
/*
    Mary Catherine Shepherd
    IS 413
    Mission 11

    Book.ts

    This TypeScript interface defines the structure of a Book object
    used in the React frontend. It matches the Book model returned
    by the ASP.NET Core API so the frontend knows the shape of the data.
*/

export interface Book {

    // Unique identifier for each book
    bookID: number

    // Title of the book
    title: string

    // Author of the book
    author: string

    // Publisher of the book
    publisher: string

    // ISBN number used to identify the book
    isbn: string

    // Category or classification of the book
    classification: string

    // Category field (synced with classification)
    category?: string

    // Number of pages in the book
    pageCount: number

    // Price of the book
    price: number
}